src/sql/base/browser/ui/panel/panel.ts

```diff
@@ -161,19 +164,40 @@ export class TabbedPanel extends Disposable {
 			invokeTabSelectedHandler();
 		}));
 
-		tab.disposables.push(DOM.addDisposableListener(tabHeaderElement, DOM.EventType.KEY_UP, (e: KeyboardEvent) => {
+		tab.disposables.push(DOM.addDisposableListener(tabHeaderElement, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
 			let event = new StandardKeyboardEvent(e);
 			if (event.equals(KeyCode.Enter)) {
 				this.showTab(tab.tab.identifier);
 				invokeTabSelectedHandler();
 				e.stopImmediatePropagation();
 			}
+			if (event.equals(KeyCode.RightArrow)) {
+				let currentIndex = this._tabOrder.findIndex(x => x === tab.tab.identifier);
+				this.focusNextTab(currentIndex + 1);
+			}
+			if (event.equals(KeyCode.LeftArrow)) {
+				let currentIndex = this._tabOrder.findIndex(x => x === tab.tab.identifier);
+				this.focusNextTab(currentIndex - 1);
+			}
+			if (event.equals(KeyCode.Tab)) {
+				e.preventDefault();
+				if (this._shownTabId) {
+					const shownTab = this._tabMap.get(this._shownTabId);
+					if (shownTab) {
+						this.setFocusToFirstElemnt(shownTab);
```

Comment:
you should just call focus on the tab object, they should define their own way of how they are focused.

Reply:
you mean shownTab.body.focus(); it wasn't focusing to next element... the tab from header should focus to next element. modal also does similar so I think this might be required...?